scripts/linters/pylint_extensions.py

```diff
@@ -1653,100 +1653,121 @@ class SingleLineCommentChecker(checkers.BaseChecker):
+        "# next_line is of string type." or "# v2 version does not have
+        ExplorationStats Model." or "# int. The file size, in bytes.".
 
-            # Ignore lines which are not comments.
-            if not line.startswith(b'#'):
-                continue
+        Args:
+            line: str. The current line of comment.
+            line_num: int. Line number of the current comment.
+        """
+        # Check if variable name is used.
+        underscore_is_present = '_' in line.split()[1]
+
+        # Check if allowed prefix is used.
+        allowed_prefix_is_present = any(
+            line[2:].startswith(word) for word in
+            self.config.allowed_comment_prefixes)
+
+        # Check if comment contains any excluded phrase.
+        excluded_phrase_is_present_at_beginning = any(
+            word in line.split()[1] for word in EXCLUDED_PHRASES)
+        if (re.search(br'^# [a-z][A-Za-z]*.*$', line) and not (
+                excluded_phrase_is_present_at_beginning or
+                underscore_is_present or allowed_prefix_is_present)):
+            self.add_message(
+                'no-capital-letter-at-beginning', line=line_num)
 
-            # Check if comment contains any excluded phrase.
-            word_is_present_in_excluded_phrases = any(
-                word in line for word in EXCLUDED_PHRASES)
+    def _check_punctuation(self, line, line_num):
+        """Checks if the comment starts with a correct punctuation.
 
-            # Comments may include a lowercase character at beginning
-            # or may not use a punctuation at end if it contains a
-            # excluded phrase e.g. "# coding: utf-8".
-            if word_is_present_in_excluded_phrases:
-                continue
+        Args:
+            line: str. The current line of comment.
+            line_num: int. Line number of the current comment.
+        """
+        excluded_phrase_is_present_at_end = any(
+            word in line for word in EXCLUDED_PHRASES)
+        # Comments must end with the proper punctuation.
+        last_char_is_invalid = line[-1] not in (
+            ALLOWED_TERMINATING_PUNCTUATIONS)
+
+        excluded_phrase_at_beginning_of_line = any(
+            line[1:].startswith(word) for word in EXCLUDED_PHRASES)
+        if (last_char_is_invalid and not (
+                excluded_phrase_is_present_at_end or
+                excluded_phrase_at_beginning_of_line)):
+            self.add_message('invalid-punctuation-used', line=line_num)
 
-            if not next_line.startswith(b'#'):
-                # Comments must end with the proper punctuation.
-                last_char_is_invalid = line[-1] not in (
-                    ALLOWED_TERMINATING_PUNCTUATIONS)
-                if last_char_is_invalid:
-                    self.add_message(
-                        'invalid-punctuation-used', line=line_num + 1)
+    def process_tokens(self, tokens):
```

Comment:
Good work on simplifying this driver function!

Reply:
Thanks :)